app/src/main/java/com/github/studydistractor/sdp/ui/CreateEventScreen.kt

```diff
@@ -0,0 +1,287 @@
+        uiState.startMinute
+    )
+    val endDateTimePicker = dateTimePickerDialog(
+        context,
+        { y, m, d, h, min -> createEventViewModel.setEndDateTime(y, m, d, h, min) },
+        uiState.endYear,
+        uiState.endMonth,
+        uiState.endDayOfMonth,
+        uiState.endHour,
+        uiState.endMinute
+    )
+
+
+
+    Box(
+        modifier = Modifier
+            .padding(16.dp)
+            .testTag("create-activity-screen__main-container")
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(16.dp)
+                .verticalScroll(rememberScrollState())
+                .testTag("create-event-screen__column"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center
+        ) {
+
+            Row {
+                androidx.compose.material.Text(
+                    text = "Create event",
+                    style = MaterialTheme.typography.titleLarge,
+                    modifier = Modifier
+                        .padding(vertical = 32.dp)
+                        .testTag("create-event-screen__title")
+                )
+            }
+
+            CreateActivityNameAndDescriptionFields(
+                createActivityViewModel = createEventViewModel
+            )
+
+            CreateDateTimeField(
+                valueText = uiState.startDateTime,
+                onValueChange = { createEventViewModel.updateStartDateTimeText(it) },
+                labelText = "Start date and time (dd-mm-yyyy hh:mm)",
+                onClickButton = { startDateTimePicker.show() }
+            )
+
+            CreateDateTimeField(
+                valueText = uiState.endDateTime,
+                onValueChange = { createEventViewModel.updateEndDateTimeText(it) },
+                labelText = "End date and time (dd-mm-yyyy hh:mm)",
+                onClickButton = { endDateTimePicker.show() }
+            )
+
+            CreateLatitudeAndLongitudeFields(
+                createActivityViewModel = createEventViewModel,
+                uiState = uiState
+            )
+
+            PointsAwardedField(createActivityViewModel = createEventViewModel, uiState = uiState)
+            LateParticipationCheckbox(
+                createEventViewModel = createEventViewModel,
+                uiState = uiState
+            )
+
+            CreateActivityButton(
+                createActivityViewModel = createEventViewModel,
+                onActivityCreated = onEventCreated,
+                buttonText = "Create event"
+            )
+        }
+    }
+}
+
+/**
+ * Creates a date picker dialog
+ * @param context Context
+ * @param setDateTime Callback to be called when date and time is set
+ * @param year Year
+ * @param month Month
+ * @param dayOfMonth Day of month
+ * @param hour Hour
+ * @param minute Minute
+ * @return DatePickerDialog
+ */
+private fun dateTimePickerDialog(
```

Comment:
Should be a composable component, with a composable name (=with an uppercase letter) such as `DateTimePickerDialog24H`

Reply:
This is a `DatePickerDialog` object from `android.app.DatePickerDialog`, it's not a composable, so this is not possible.

Reply:
Mmmh, actually if you just add the @Composable decorator, it works?

Reply:
I guess, but it doesn't make it a composable : P